requirements.txt

```diff
@@ -8,7 +8,4 @@ dill>=0.3
 python-dateutil>=2.8.0
 stevedore>=3.0.0
 typing-extensions; python_version<'3.11'
-# symengine pinning needed due lowered precision handling complex
-# multiplication in version 0.10 wich breaks parameter assignment test
-# (can be removed once issue is fix)
-symengine>=0.9, <0.10; platform_machine == 'x86_64' or platform_machine == 'aarch64' or platform_machine == 'ppc64le' or platform_machine == 'amd64' or platform_machine == 'arm64'
+symengine>=0.9, <0.10
```

Comment:
should we keep the comment explaining why we are pinning symengine to <0.10?

Reply:
Yeah, normally I'd agree and add it back. But, https://github.com/Qiskit/qiskit/pull/11262 uncaps the symengine version. So I don't think it matters too much anymore.